packages/@uppy/aws-s3/types/index.d.ts

```diff
@@ -10,8 +9,23 @@ export interface AwsS3UploadParameters {
     headers?: { [type: string]: string }
 }
 
-export interface AwsS3Options extends AwsS3MultipartOptions {
-    /** @deprecated future versions of this plugin will use the Expires value from the backend */
+export interface AwsS3Options extends PluginOptions {
+    companionUrl?: string | null
+    companionHeaders?: Record<string, string>
+    allowedMetaFields?: Array<string> | null
+    getUploadParameters?: (file: UppyFile) => MaybePromise<{
+      method?: 'POST'
+      url: string
+      fields?: Record<string, string>
+      headers?: Record<string, string>
+    } | {
+      method: 'PUT'
+      url: string
```

Comment:
If `fields` are not allowed I think we need this?
```suggestion
      url: string
      fields: never
```

Reply:
That is a better type - I have updated to using a 'never'